buffer/src/main/java/io/netty/buffer/PoolChunk.java

```diff
@@ -223,11 +223,19 @@ private int usage(int freeBytes) {
     }
 
     boolean allocate(PooledByteBuf<T> buf, int reqCapacity, int normCapacity) {
+        final long handle;
         if ((normCapacity & subpageOverflowMask) != 0) { // >= pageSize
-            return allocateRun(buf, reqCapacity, normCapacity);
+            handle =  allocateRun(normCapacity);
         } else {
-            return allocateSubpage(buf, reqCapacity, normCapacity);
+            handle = allocateSubpage(normCapacity);
         }
+
+        if (handle > 0) {
```

Comment:
This was effectively `>= 0` before, not sure if it makes a difference. Might also read slightly simpler if inverted i.e. initBuf outside the if block

Reply:
fixed